ast/dml.go

```diff
@@ -1519,7 +1519,41 @@ type FrameBound struct {
 
 // Restore implements Node interface.
 func (n *FrameBound) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if n.UnBounded {
+		ctx.WriteKeyWord("UNBOUNDED")
+	}
+	switch n.Type {
+	case CurrentRow:
+		ctx.WriteKeyWord("CURRENT ROW")
+	case Preceding, Following:
+		if n.Unit != nil {
+			ctx.WriteKeyWord("INTERVAL ")
+		}
+		if n.Expr != nil {
+			if err := n.Expr.Restore(ctx); err != nil {
+				return errors.Annotate(err, "An error occurred while restore FrameBound.Expr")
+			}
+		}
+		if n.Unit != nil {
+			// Here the Unit string should not be quoted
+			singleQuotesFlag := ctx.Flags & RestoreStringSingleQuotes
+			doubleQuotesFlag := ctx.Flags & RestoreStringDoubleQuotes
+			ctx.Flags &= ^RestoreStringSingleQuotes
+			ctx.Flags &= ^RestoreStringDoubleQuotes
```

Comment:
This doesn't seem like the correct approach. For instance, the following function call shouldn't parse either:

```sql
select date_add(current_timestamp, interval 1 'year');
```

which means the same logic got to be duplicated for FuncCallExpr.

----

I have no idea why `Unit` is considered a *string literal* expression?! It should be made a special "unit expression" based on an enumerated constant. That involves changing TiDB itself though.

Reply:
Yes, I think it is reasonable. And without this change, it is harder for `FuncCallExpr` to handle this kind of cases.